src/primitives/torus.jl

```diff
@@ -0,0 +1,67 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Torus(center, normal, major, minor)
+
+A torus centered at `center` with axis of revolution directed by 
+`normal` and with radii `major` and `minor`. 
+
+"""
+struct Torus{T} <: Primitive{3,T}
+  center::Point{3,T}
+  normal::Vec{3,T}
+  major::T
+  minor::T
+end
+
+function Torus(center::Tuple, normal::Tuple, major::T1, minor::T2) where {T1,T2}
+  T = promote_type(eltype(center), eltype(normal), T1, T2, Float32)  
+  Torus(Point(T.(center)), Vec(T.(normal)), T(major), T(minor))
+end
```

Comment:
Please follow the pattern in Ball, i.e., first a constructor with Point and Vec and type parameter T, then a second constructor with Tuple.